### ISSUE TITLE
[native] branch v2.0.0-rc1, bug for multiple producer instances in a same process?

### ISSUE THREAD

**Issue by plhwin** [apache/rocketmq-client-go#381]
Please tell us about your environment:

> What is your OS? 
>  macOS Catalina 10.15.2
> 
> What is your client version? 
> rocketmq-client-go branch v2.0.0-rc1 (branch native got the same result)
> 
> What is your RocketMQ version?  
> V4_6_0

I simulated a typical quote sending scenario in a FinTech scenario: receive multiple liquidity data at the same time (each liquidity has multiple different symbols), and then use liquidity + symbol as the sharding key to send to TopicTest. the topic of TopicTest has 16 queues.

the program simulation:
1. a separate producer instance for each liquidity
2. each liquidity sends partition sequence message to rocketmq in a separate goroutine

but, it seem some error happend with SendOneWay(or SendSync) methods, the error was: 「 send error: the topic=TopicTest route info not found」

here is the example code below:
[https://play.golang.org/p/jxwX332Z8uk](https://play.golang.org/p/jxwX332Z8uk)

the program accepts the parameters like below:
go run main.go -limit=10 -liquidity=L1,L2,L3 -symbol=S1   

the execution results of some different parameters are shown below

![image](https://user-images.githubusercontent.com/1219345/72038885-f783e980-32dd-11ea-9032-383eb729b743.png)
`go run main.go -limit=10 -liquidity=L1 -symbol=S1 `
with only one liquidity L1, it mean only one producer instance, the results are as expected, all sent successfully

![image](https://user-images.githubusercontent.com/1219345/72038969-47fb4700-32de-11ea-808e-152ef5b6ac83.png)
`go run main.go -limit=10 -liquidity=L1,L2 -symbol=S1`

with two producer instance, only L2 success, L1 all failed.

![image](https://user-images.githubusercontent.com/1219345/72039012-6c572380-32de-11ea-955c-6d824fabc491.png)
`go run main.go -limit=10 -liquidity=L1,L2,L3 -symbol=S1`
with 3 producer instance, only L3 success, L1 and L2 all failed.

in summary, always the last producer instance succeed

![image](https://user-images.githubusercontent.com/1219345/72039069-a7f1ed80-32de-11ea-9f5b-dc204a0f8043.png)
`go run main.go -limit=10 -liquidity=L1 -symbol=S1,S2,S3`
with 1 liquidity producer instance + 3 symbol under this liquidity, all success

**Comment by plhwin** [comment]
found the reason:
to use multiple producer instances in a process, should be specify a different name for each instance

use `producer.WithInstanceName("name")` to specify a instance name
